src/Compilers/CSharp/Portable/Syntax/SyntaxExtensions.cs

```diff
@@ -387,5 +387,61 @@ internal static bool IsOutVarDeclaration(this DeclarationExpressionSyntax p)
         {
             return p.Designation.Kind() == SyntaxKind.SingleVariableDesignation && p.IsOutDeclaration();
         }
+
+        /// <summary>
+        /// Visits all the ArrayRankSpecifiers of a typeSyntax, invoking an action on each one in turn.
+        /// </summary>
+        /// <param name="type"></param>
+        /// <param name="action"></param>
+        /// <param name="argument">The argument that is passed to the action whenever it is invoked</param>
+        internal static void VisitRankSpecifiers<TArg>(this TypeSyntax type, Action<ArrayRankSpecifierSyntax, TArg> action, TArg argument)
+        {
+            switch (type)
```

Comment:
>type [](start = 20, length = 4)

For many cases, switching on an enum is faster.  You can accomplish that by writing
``` c#
switch (type.Kind(), type)
{
    case (SyntaxKind.ArrayType, ArrayTypeSyntax arrayTypeSyntax):
        ...
    case (SyntaxKind.NullableType, NullableTypeSyntax nullableTypeSyntax):
        ....
}
```
And so on.  Please modify this to switch on the kind field.  For only two or three cases, switching directly on the type is fine.

Reply:
That syntax doesn't seem to be allowed, even in master:

https://sharplab.io/#v2:EYLgtghgzgLgpgJwD4AEAMACFBGA3AWACgUBmLAJgwGEMBvIjRrMlAFgwFkAKASjoaaCAbhAQZgGALwYAdnADuGAPLAAVnADGMXgUKDBUeQEsYGgBYYuXYADoA4nBgAJaGaoB7ACZxeAGnE8PAL6/Hoh4Ywa0HCW2P4AyjAIRjIA5hiwCDwgwREhKADsGEkArnC6EQC+udWElUA=